src/main/java/rocket/task/TaskList.java

```diff
@@ -0,0 +1,67 @@
+package rocket.task;
+
+import rocket.exception.RocketRuntimeException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public Task get(int index) {
+        if (index < 0 || index >= tasks.size()) {
+            throw new ArrayIndexOutOfBoundsException("Index: " + index + ", Size: " + tasks.size());
+        }
+        Task task = tasks.get(index);
+        String name = task.getName();
+        boolean mark = task.getMark();
+        TaskType taskType = task.getType();
+        switch (taskType) {
+            case TODO:
+                return new Todo(name, mark);
+            case DEADLINE:
+                LocalDate by = task.getBy();
+                return new Deadline(name, mark, by);
+            case EVENT:
+                LocalDate from = task.getFrom();
+                LocalDate to = task.getTo();
+                return new Event(name, mark, from, to);
+            default:
+                throw new RocketRuntimeException("Failed to get Task from TaskList");
```

Comment:
I think you may want to remove case indents and add a fall through comment since break was ommited from your switch case as per our briefing notes.

```suggestion
        switch (taskType) {
        case TODO:
            return new Todo(name, mark);
        case DEADLINE:
            LocalDate by = task.getBy();
            return new Deadline(name, mark, by);
        case EVENT:
            LocalDate from = task.getFrom();
            LocalDate to = task.getTo();
            return new Event(name, mark, from, to);
        default:
            throw new RocketRuntimeException("Failed to get Task from TaskList");
            // Add Fall Through Comment
```